easybuild/tools/github.py

```diff
@@ -1591,6 +1591,26 @@ def update_pr(pr_id, paths, ecs, commit_msg=None):
     print_msg(msg, log=_log)
 
 
+def check_online_status():
+    """
+    Check whether we currently are online
+    Return True if online, else an URLError instance of the last failure
+    """
+    # Try repeatedly and with different URLs to cater for flaky servers
+    # E.g. Github returned "HTTP Error 403: Forbidden" and "HTTP Error 406: Not Acceptable" randomly
+    # Timeout and repeats set to total 1 minute
+    urls = [GITHUB_URL, GITHUB_API_URL]
+    num_repeats = 6
+    for attempt in range(num_repeats):
+        try:
+            urlopen(urls[attempt % len(urls)], timeout=10)
+            result = True
+            break
+        except URLError as err:
+            result = err
```

Comment:
I hate to be pedantic, but we're only keeping track of the last failure here, how about keeping track of al off them (incl. the URLs we tried)?

So, make `result` a list of error messages in case of failures, and mention the URL that was tested in there...

Reply:
You really want to see all 6 error messages and the 2 URLs repeated 3 times each? Sounds overkill to me.

I can do that, but not til Monday. So maybe merge this and hope we never see an error again?

Reply:
Well, it just feels wrong to hide the earliest 5 errors...
Are they all the same? Is it a different issue on every attempt? Is only one of the URLs broken?

Those are all questions you may want an answer to when the check fails after 6 attempts.

Reply:
Current test will look like this (prefixed urls with "fo://" to force failure):

> Making sure we're online...FAIL (fo://https://api.github.com: <urlopen error unknown url type: fo>, fo://https://github.com: <urlopen error unknown url type: fo>)

Reply:
See https://github.com/easybuilders/easybuild-framework/pull/3194/commits/2e11b4829a6c804750a954a13846730b552d612b